acasclient/selfile.py

```diff
@@ -0,0 +1,851 @@
+"""
+A module to read, edit, and write ACAS 'Single Experiment
+Loader' files.
+Common formats are supported for both input and output: csv, and
+excel
+"""
+# Contributors:  Shawn Watts
+
+
+################################################################################
+# Packages
+################################################################################
+import collections
+import logging
+import io
+import os
+from functools import partial
+import pandas as pd
+
+
+################################################################################
+# Globals/Constants
+################################################################################
+# File formats/types.
+CSV = '.csv'
+XLS = '.xls'
+XLSX = '.xlsx'
+LOT_CORP_NAME = 'Corporate Batch ID'
+GENERIC = 'Generic'
+DOSE_RESPONSE = 'Dose Response'
+
+# ACAS-CmpdReg constants in sdf files.
+PARENT_CORP_NAME = 'Parent Corp Name'
+PARENT_STEREO_CAT = 'Parent Stereo Category'
+PARENT_ALIASES = 'Parent Aliases'
+LOT_NUM = 'Lot Number'
+LOT_NOTEBOOK_PAGE = 'Lot Notebook Page'
+LOT_CHEMIST = 'Lot Chemist'
+LOT_DATE = 'Lot Synthesis Date'
+LOT_COMMENTS = 'Lot Comments'
+LOT_SUPPLIER = 'Lot Supplier'
+LOT_SUPPLIER_ID = 'Lot Supplier ID'
+LOT_BARCODE = 'Lot Barcode'
+LOT_AMOUNT = 'Lot Amount'
+LOT_AMOUNT_UNITS = 'Lot Amount Units'
+LOT_PURITY = 'Lot Purity'
+LOT_PURITY_MEASURED_BY = 'Lot Purity Measured By'
+LOT_SALT_ABBREV = 'Lot Salt Abbrev'
+LOT_SALT_EQ = 'Lot Salt Equivalents'
+REG_LOT_CORP_NAME = 'Registered Lot Corp Name'
+REG_PARENT_CORP_NAME = 'Registered Parent Corp Name'
+REG_LOT_AMOUNT =  'Registered Lot Amount'
+REG_LOT_AMOUNT_UNITS =  'Registered Lot Amount Units'
+REG_PARENT_ALIASES = 'Registered Parent Aliases'
+REG_LEVEL = 'Registration Level'
+
+
+################################################################################
+# Logging
+################################################################################
+logger = logging.getLogger(__name__).addHandler(logging.NullHandler())
```

Comment:
Changed logger as per:https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library